tutorons/python/views.py

```diff
@@ -0,0 +1,71 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+import logging
+import json
+from django.views.decorators.csrf import csrf_exempt
+from django.template.loader import get_template
+from django.template import Context
+from django.http import HttpResponse
+
+from tutorons.common.htmltools import HtmlDocument
+from tutorons.common.util import log_region, package_region
+from tutorons.common.scanner import NodeScanner
+from tutorons.python.detect import PythonBuiltInExtractor
+from tutorons.python.explain import explain as python_explain
+from tutorons.python.render import render as python_render
+
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+region_logger = logging.getLogger('region')
+
+
+@csrf_exempt
+def scan(request):
+
+    doc_body = request.POST.get('document')
+    origin = request.POST.get('origin')
+    region_logger.info("Request for page from origin: %s", origin)
+
+    explained_regions = []
+    document = HtmlDocument(doc_body)
+    builtin_extractor = PythonBuiltInExtractor()
+
+    builtin_scanner = NodeScanner(builtin_extractor, ['code', 'pre'])
+    regions = builtin_scanner.scan(document)
+    for r in regions:
+        log_region(r, origin)
+        hdr, exp, url = python_explain(r.string)
+        document = python_render(r.string, hdr, exp, url)
+        explained_regions.append(package_region(r, document))
+
+    return HttpResponse(json.dumps(explained_regions, indent=2))
+
+
+@csrf_exempt
+def explain(request):
```

Comment:
Hi Matthew!  You can right-click on selected text, and choose the option to "explain as <language>" from the Tutorons context menu:

<img width="472" alt="tutorons-selection" src="https://cloud.githubusercontent.com/assets/2358524/12226874/d1205ca4-b7d7-11e5-9285-79e1ce245b24.png">

See line 85 of `addon/lib/main.js` in the tutorons-library repo for where the "explain as" actions are defined for each language and added to the browser's context menu.  You'll have to add an action for Python in this file ;)